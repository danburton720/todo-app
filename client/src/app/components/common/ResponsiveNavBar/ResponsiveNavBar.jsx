import React from 'react';
import styles from './responsiveNavBar.scss';
import IconButton from '../IconButton/IconButton';
import {Menu, X} from 'react-feather';

export default class ResponsiveNavBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            showSideMenu: false
        };
        this.toggleSideMenu = this.toggleSideMenu.bind(this);
    }

    componentDidUpdate(prevProps, prevState) {
        if(this.state.showSideMenu && !prevState.showSideMenu) {
            document.getElementById('sideBar').className = 'sideBar show';
        }
        if(!this.state.showSideMenu && prevState.showSideMenu) {
            document.getElementById('sideBar').className = 'sideBar';
        }
    }

    toggleSideMenu() {
        this.setState({
            showSideMenu: !this.state.showSideMenu
        });
    }

    render() {
        return (
            <div className={styles.responsiveNavBar}>
                <IconButton
                    icon={this.state.showSideMenu ? X : Menu}
                    onClick={this.toggleSideMenu}
                />
            </div>
        )
    }
}