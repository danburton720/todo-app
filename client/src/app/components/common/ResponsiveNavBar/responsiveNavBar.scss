.responsiveNavBar {
  display: none;
  .iconButton {
    padding: 20px;
  }
}

@media (max-width: 800px) {
  .responsiveNavBar {
    display: flex;
  }
}