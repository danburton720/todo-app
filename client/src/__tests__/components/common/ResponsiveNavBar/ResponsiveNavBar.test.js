import React from 'react';
import {shallow} from 'enzyme';
import ResponsiveNavBar from '../../../../app/components/common/ResponsiveNavBar/ResponsiveNavBar';

describe('ResponsiveNavBar', () => {
    let shallowWrapper;
    let props;
    const wrapper = () => {
        if (!shallowWrapper) {
            shallowWrapper = shallow(<ResponsiveNavBar {...props}/>);
        }
        return shallowWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowWrapper = undefined;
    });

    describe('render', () => {
        it('should return an IconButton with an Menu icon', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return an IconButton with an X icon', () => {
            const getElementByIdSpy = jest.spyOn(document, 'getElementById');
            let className = 'sideBar';
            getElementByIdSpy.mockImplementation(() => ({
                className
            }));
            wrapper().find('IconButton').first().props().onClick();
            expect(wrapper()).toMatchSnapshot();
        });
    });
});