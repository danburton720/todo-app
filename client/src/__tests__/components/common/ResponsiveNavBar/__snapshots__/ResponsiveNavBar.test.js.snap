// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveNavBar render should return an IconButton with an Menu icon 1`] = `
<div
  className="responsiveNavBar"
>
  <IconButton
    icon={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Menu",
        "propTypes": Object {
          "color": [Function],
          "size": [Function],
        },
        "render": [Function],
      }
    }
    onClick={[Function]}
    size={18}
  />
</div>
`;

exports[`ResponsiveNavBar render should return an IconButton with an X icon 1`] = `
<div
  className="responsiveNavBar"
>
  <IconButton
    icon={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "X",
        "propTypes": Object {
          "color": [Function],
          "size": [Function],
        },
        "render": [Function],
      }
    }
    onClick={[Function]}
    size={18}
  />
</div>
`;
